common/src/main/java/com/wynntils/features/tooltips/ItemCompareFeature.java

```diff
@@ -4,162 +4,590 @@
+            return gearItem.getGearType() == gearItemReference.getGearType();
+        }
+        return false;
     }
 
-    private void onCompareModeKeyPress(Slot hoveredSlot) {
-        if (comparedItem != null) {
-            comparedItem = null;
-            compareToEquipped = true;
-        } else {
-            compareToEquipped = !compareToEquipped;
-        }
+    private boolean isMatchingType(ItemStack itemStack, GearTypeItemProperty gearItemReference) {
+        Optional<GearTypeItemProperty> gearOpt = Models.Item.asWynnItemProperty(itemStack, GearTypeItemProperty.class);
+        return gearOpt.isEmpty() ? false : gearOpt.get().getGearType() == gearItemReference.getGearType();
     }
 
     private void onSelectKeyPress(Slot hoveredSlot) {
         if (hoveredSlot == null) return;
+        // Selecting items behaves weirdly on tm updates, so no selecting in tm
+        if (Models.Container.getCurrentContainer() instanceof TradeMarketContainer) return;
+        int slot = hoveredSlot.getContainerSlot();
+
+        // Selecting armor and accessory slots is not allowed
+        if (slot == 9 || slot == 10 || slot == 11 || slot == 12 || slot == 36 || slot == 37 || slot == 38 || slot == 39)
+            return;
 
         ItemStack itemStack = hoveredSlot.getItem();
-        Optional<GearItem> wynnItemOpt = Models.Item.asWynnItem(itemStack, GearItem.class);
+        if (itemStack.isEmpty()) return;
+
+        Optional<WynnItem> wynnItemOpt = Models.Item.getWynnItem(itemStack);
         if (wynnItemOpt.isEmpty()) return;
+        WynnItem wynnItem = wynnItemOpt.get();
+        if (!(wynnItem instanceof GearItem)) return;
 
-        if (comparedItem == itemStack) {
-            comparedItem = null;
+        if (selectedItems.containsKey(wynnItem)) {
+            selectedItems.remove(wynnItem);
         } else {
-            comparedItem = itemStack;
-            compareToEquipped = false;
+            selectedItems.put(wynnItem, itemStack);
+        }
+    }
+
+    private String getTag(String key) {
+        return I18n.get(key);
+    }
+
+    private List<Component> getWynnOrVanillaLines(
+            AbstractContainerScreen<?> screen, WynnItem wynnItem, ItemStack itemStack) {
+        List<Component> wynnTooltip = TooltipUtils.getWynnItemTooltip(itemStack, wynnItem);
+        return wynnTooltip.isEmpty() ? screen.getTooltipFromItem(McUtils.mc(), itemStack) : wynnTooltip;
+    }
+
+    private MutableComponent getPaddedComponent(String string, int tooltipWidth) {
+        Font font = FontRenderer.getInstance().getFont();
+        // This is not the exact center, but good enough
+        int count = Math.round((tooltipWidth - font.width(string)) / 2.0f / font.width(" "));
+        return Component.literal(" ".repeat(count) + string);
+    }
+
+    private void removeFlavourText(List<Component> lines) {
+        while (lines.getLast().getString().startsWith("§8")) {
+            lines.removeLast();
+        }
+    }
```

Comment:
I'd like it so you refactor this method to use StyledText and check the first components actual ChatFormatting. I fear this could easily break if we changed how the lore component looks like upstream, `Component#getString` is not known to be the most stable result producing method..

Reply:
I'm having a bit of a problem doing this. In particular, I don't understand why lore lines have style={color=dark_purple,!italic}

Reply:
Weird, I can take a look in a bit if you haven't figured it by then.

Reply:
I figured out that `GearTooltipComponent#buildFooterTooltip` is only used to build tooltips for fake items like ones that are displayed in Item Guide. So I guess lore having `dark_purple` as color in Style is some error(?) on Wynncraft's side